src/audio/wav64_opus.c

```diff
@@ -97,17 +98,18 @@ static void waveform_opus_read(void *ctx, samplebuffer_t *sbuf, int wpos, int wl
             memset(out, 0, st->xhead.frame_size * wav->wave.channels * sizeof(int16_t));
         } else {
             // Read frame size
-            int nb = io_read16(st->current_rom_addr);
-            assertf(nb <= st->xhead.max_cmp_frame_size, "opus frame size too large: %d (%ld)", nb, st->xhead.max_cmp_frame_size);
+            uint16_t nb = 0;
+            read(st->current_fd, &nb, 2);
+            assertf(nb <= st->xhead.max_cmp_frame_size, "opus frame size too large: %08X (%ld) %d", nb, st->xhead.max_cmp_frame_size, callcount);
 
-            // Read frame
-            data_cache_hit_writeback_invalidate(buf, nb);
-            dma_read(buf, st->current_rom_addr + 2, nb);
+            unsigned long aligned_frame_size = nb; 
+            if (aligned_frame_size & 1) {
+                aligned_frame_size += 1;
+            }
 
-            // Update ROM pointer after current frame
-            st->current_rom_addr += nb + 2;
-            if (st->current_rom_addr & 1)
-                st->current_rom_addr += 1;
+            // Read frame
+            data_cache_hit_writeback_invalidate(buf, aligned_frame_size);
+            read(st->current_fd, buf, aligned_frame_size);
```

Comment:
You need to increase the size of `buf` by 1, otherwise this can cause a buffer overflow.